File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):

    ls = [t[i:i+len(p)] for i in range(len(t) - len(p) + 1)]
    # i = str(int(p) + 1)
    # ls.append(i)
    # ls.sort()
    # print(i,p,ls)
    # return ls.index(i)

    answer = 0
    for st in ls:
        if int(st) <= int(p):
            answer +=1
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>
#include <vector>
 #include <cstring>
using namespace std;


int solution(string t, string p)
{

    int answer = 0;
     long long str_size = p.size();

    string str;

    for ( long long i = 0; i <= t.size() - str_size; i++)
    {
        for( long long j = 0; j < str_size; j++)
        {

            if (t[i + j] == NULL)
            {

            }
            else
            {
                str += t[i + j];
            }   
        }

        if (str.size() == str_size)
        {
            if ((str) <= (p))
            {
                answer++;
            }

            str = "";
        }
    }
    return answer;
}